### ISSUE TITLE
USB shared network, ALVR can't connect

### ISSUE THREAD

**Issue by admin667788** [alvr-org/ALVR#2271]
I was using the phone's USB sharing network to connect to the PICO VR glasses. The PICO WIFI was turned off, and the traffic was routed from the phone's USB sharing network. The phone and computer are in a LAN, and the phone opens a USB shared network connection to PICO. Install ALVR apk on PICO, after startup, the computer also starts, PICO cannot enter the Steam interface. How do I solve this

**Comment by The-personified-devil** [comment]
Tbh I don't understand your explanation. Can you explain it in terms of `X connected to Y via Z`, so `Computer connected to Phone via Wifi` for all the different connections please?

**Comment by admin667788** [comment]
![image](https://github.com/user-attachments/assets/c0ed1976-2a02-457e-af59-f067930aef7e)
After connecting to the PICO glasses, the phone opens the USB tethering, PICO network from the phone, the phone and VR glasses set up a subnet. Now PICO cannot launch ALVR after being unable to link to Steam

**Comment by admin667788** [comment]
> Tbh I don't understand your explanation. Can you explain it in terms of `X connected to Y via Z`, so `Computer connected to Phone via Wifi` for all the different connections please?
The phone and computer are connected to the LAN, the phone and PICO group a subnet, the computer can ping the phone, but can not find PICO.
Router IP: 192.168.1.1, mobile phone LAN IP: 192.168.1.171, computer LAN IP: 192.168.1.8,
The phone connects to PICO via USB tethering, and PICO's IP is 192.168.254.147.
After both the ALVR computer and PICO are started, PICO under the subnet: 192.168.254.147 cannot be found.

**Comment by admin667788** [comment]
@The-personified-devil 
The phone and computer are connected to the LAN, the phone and PICO group a subnet, the computer can ping the phone, but can not find PICO.
Router IP: 192.168.1.1, mobile phone LAN IP: 192.168.1.171, computer LAN IP: 192.168.1.8,
The phone connects to PICO via USB tethering, and PICO's IP is 192.168.254.147.
After both the ALVR computer and PICO are started, PICO under the subnet: 192.168.254.147 cannot be found.

**Comment by The-personified-devil** [comment]
You could try looking into this: https://github.com/alvr-org/ALVR/wiki/Headset-and-ALVR-streamer-on-separate-networks
But apart from that this is simply a configuration issue with subnets behaving as they should, so I'm closing this issue. If you want some further help asking in the discord (or matrix) is probably your best bet

**Comment by admin667788** [comment]
@The-personified-devil This scheme is not suitable. Can PICO find the Steam server? Instead of the server looking for PICO.

**Comment by The-personified-devil** [comment]
@admin667788 the function of subnets is to ensure that the devices inside of different subnets don't see each other. That means that no scheme of local network discovery will work. And since latency is critical and the safety of managing access to devices is also critical only local network connections are sensible. Especially since those can still be routed across the internet by third party software, if needed.
